lib/fog/core/model.rb

```diff
@@ -20,6 +21,10 @@ def initialize(new_attributes = {})
       merge_attributes(attribs)
     end
 
+    def cache
+      Fog::Cache.new(self)
+    end
```

Comment:
The only method that is available from a `Fog::Model` is via this injected Cache object, and the only instance methods that exist are `model.cache.dump` and `model.cache.dump_to`. Everything else is a Cache class method so not available via this.